src/middleware/jira-symmetric-jwt-middleware.ts

```diff
@@ -92,7 +96,16 @@ const verifySymmetricJwt = async (req: Request, token: string, installation: Ins
 
 	try {
 		const claims = decodeSymmetric(token, secret, algorithm, false);
-		const tokenType = checkPathValidity(req.originalUrl) && req.method == "GET" || checkGenericContainerActionUrl(`${envVars.APP_URL}${req.originalUrl}`)? TokenType.normal : TokenType.context;
+		let tokenType;
+
+		if (await booleanFlag(BooleanFlags.ENABLE_GENERIC_CONTAINERS)) {
+			tokenType = checkPathValidity(req.originalUrl) && req.method == "GET"
+				|| await checkGenericContainerActionUrl(`${envVars.APP_URL}${req.originalUrl}`) ? TokenType.normal
+				: TokenType.context;
+		} else {
+			tokenType = checkPathValidity(req.originalUrl) && req.method == "GET" ? TokenType.normal : TokenType.context;
+		}
```

Comment:
/nit i find below more readable

```
const getTokenType = async (url: string, method: string): Promise<TokenType> => {
  if (checkPathValidity(url) && method === "GET") {
    return TokenType.Normal;
  }

  if (await booleanFlag(BooleanFlags.ENABLE_GENERIC_CONTAINERS)) {
    const isGenericContainerActionUrl = await checkGenericContainerActionUrl(`${envVars.APP_URL}${url}`);
    return isGenericContainerActionUrl ? TokenType.Normal : TokenType.Context;
  }

  return TokenType.Context;
};
```

Reply:
Happy to pull this out into a function but the suggested logic won't work. If `checkPathValidity(req.originalUrl) && req.method == "GET"` returns true we return TokenType.normal but otherwise we need to return TokenType.context. We also can't have the FF as a second check otherwise, for every GET request, even if the FF is on, we'll never hit the second condition.